src/base/utils/compare.h

```diff
@@ -31,7 +31,10 @@
 #include <Qt>
 #include <QtSystemDetection>
 
-#if !defined(Q_OS_WIN) && (!defined(Q_OS_UNIX) || defined(Q_OS_MACOS) || defined(QT_FEATURE_icu))
+// for QT_FEATURE_xxx, see: https://wiki.qt.io/Qt5_Build_System#How_to
+#include <QtCore/private/qtcore-config_p.h>
+
+#if (defined(QT_FEATURE_icu) || defined(Q_OS_MACOS))
```

Comment:
IIRC, you should either compare `QT_FEATURE_ICU` with `1`:
```suggestion
#if ((QT_FEATURE_ICU == 1) || defined(Q_OS_MACOS))
```
or use `QT_CONFIG` macro:
```suggestion
#if (defined(Q_OS_MACOS) || QT_CONFIG(icu))
```

Reply:
You could consider how some macro is defined in Qt headers:
```c++
/*
    The QT_CONFIG macro implements a safe compile time check for features of Qt.
    Features can be in three states:
        0 or undefined: This will lead to a compile error when testing for it
        -1: The feature is not available
        1: The feature is available
*/
#define QT_CONFIG(feature) (1/QT_FEATURE_##feature == 1)
#define QT_REQUIRE_CONFIG(feature) Q_STATIC_ASSERT_X(QT_FEATURE_##feature == 1, "Required feature " #feature " for file " __FILE__ " not available.")

```

Reply:
>IIRC, you should either compare QT_FEATURE_ICU with 1:

Done. 

>QT_CONFIG(icu)

It will generate a divide by 0 compile error when icu is not available. Not sure why Qt designed it this way.

Reply:
> > QT_CONFIG(icu)
> 
> It will generate a divide by 0 compile error when icu is not available. Not sure why Qt designed it this way.

I believe you take it incorrectly. IIRC, `QT_CONFIG` will generate a divide by 0 compile error when you test for unknown/non-existent feature. Otherwise you won't be able to compile Qt with some disabled feature.